# -- coding:utf-8 --
"""
User: ww
Version: 3.8
Date: 2022/8/10 12:08
File: list_2400.py
"""
from parse.exception.exception_visa.exceptions.exception_2400 import *

List_2400 = {
    'Other': Exception_Other_error,
    '-440': Exception_Query_UNTERMINATED_after_indefinite_response,
    '-430': Exception_Query_DEADLOCKED,
    '-420': Exception_Query_UNTERMINATED,
    '-410': Exception_Query_INTERRUPTED,
    '-363': Exception_Input_buffer_overrun,
    '-362': Exception_Framing_error_in_program_message,
    '-361': Exception_Parity_error_in_program_message,
    '-360': Exception_Communications_error,
    '-350': Exception_Queue_overflow,
    '-330': Exception_Self_test_failed,
    '-314': Exception_Save_or_recall_memory_lost,
    '-315': Exception_Configuration_memory_lost,
    '-285': Exception_Program_syntax_error,
    '-284': Exception_Program_currently_running,
    '-282': Exception_Illegal_program_name,
    '-281': Exception_Cannot_create_program,
    '-260': Exception_Expression_error,
    '-241': Exception_Hardware_missing,
    '-230': Exception_Data_corrupt_or_stale,
    '-225': Exception_Out_of_memory,
    '-224': Exception_Illegal_parameter_value,
    '-223': Exception_Too_much_data,
    '-222': Exception_Parameter_data_out_of_range,
    '-221': Exception_Settings_conflict,
    '-220': Exception_Parameter_error,
    '-215': Exception_Arm_deadlock,
    '-214': Exception_Trigger_deadlock,
    '-213': Exception_Init_ignored,
    '-212': Exception_Arm_ignored,
    '-211': Exception_Trigger_ignored,
    '-210': Exception_Trigger_error,
    '-202': Exception_Settings_lost_due_to_rtl,
    '-201': Exception_Invalid_while_in_local,
    '-200': Exception_Execution_error,
    '-178': Exception_Expression_data_not_allowed,
    '-171': Exception_Invalid_expression,
    '-170': Exception_Expression_error,
    '-168': Exception_Block_data_not_allowed,
    '-161': Exception_Invalid_block_data,
    '-160': Exception_Block_data_error,
    '-158': Exception_String_data_not_allowed,
    '-154': Exception_String_too_long,
    '-151': Exception_Invalid_string_data,
    '-150': Exception_String_data_error,
    '-148': Exception_Character_data_not_allowed,
    '-144': Exception_Character_data_too_long,
    '-141': Exception_Invalid_character_data,
    '-140': Exception_Character_data_error,
    '-128': Exception_Numeric_data_not_allowed,
    '-124': Exception_Too_many_digits,
    '-123': Exception_Exponent_too_large,
    '-121': Exception_Invalid_character_in_number,
    '-120': Exception_Numeric_data_error,
    '-114': Exception_Header_suffix_out_of_range,
    '-113': Exception_Undefined_header,
    '-112': Exception_Program_mnemonic_too_long,
    '-111': Exception_Header_separator_error,
    '-110': Exception_Command_header_error,
    '-109': Exception_Missing_parameter,
    '-108': Exception_Parameter_not_allowed,
    '-105': Exception_GET_not_allowed,
    '-104': Exception_Data_type_error,
    '-103': Exception_Invalid_separator,
    '-102': Exception_Syntax_error,
    '-101': Exception_Invalid_character,
    '-100': Exception_Command_error,
    '100': Exception_Limit_1_failed,
    '101': Exception_Low_limit_2_failed,
    '102': Exception_High_limit_2_failed,
    '103': Exception_Low_limit_3_failed,
    '104': Exception_High_limit_3_failed,
    '105': Exception_Active_limit_tests_passed,
    '106': Exception_Reading_available,
    '107': Exception_Reading_overflow,
    '108': Exception_Buffer_available,
    '109': Exception_Buffer_full,
    '110': Exception_Limit_4_failed,
    '111': Exception_OUTPUT_enable_asserted,
    '112': Exception_Temperature_limit_exceeded,
    '113': Exception_Voltage_limit_exceeded,
    '114': Exception_Source_in_compliance,
    '200': Exception_Operation_complete,
    '300': Exception_Device_calibrating,
    '303': Exception_Device_sweeping,
    '305': Exception_Waiting_in_trigger_layer,
    '306': Exception_Waiting_in_arm_layer,
    '310': Exception_Entering_idle_layer,
    '408': Exception_Questionable_Calibration,
    '414': Exception_Command_Warning,
    '500': Exception_Date_of_calibration_not_set,
    '501': Exception_Next_date_of_calibration_not_set,
    '502': Exception_Calibration_data_invalid,
    '503': Exception_DAC_calibration_overflow,
    '504': Exception_DAC_calibration_underflow,
    '505': Exception_Source_offset_data_invalid,
    '506': Exception_Source_gain_data_invalid,
    '507': Exception_Measurement_offset_data_invalid,
    '508': Exception_Measurement_gain_data_invalid,
    '509': Exception_Not_permitted_with_cal_locked,
    '510': Exception_Not_permitted_with_cal_un_locked,
    '601': Exception_Reading_buffer_data_lost,
    '602': Exception_GPIB_address_lost,
    '603': Exception_Power_on_state_lost,
    '604': Exception_DC_calibration_data_lost,
    '605': Exception_Calibration_dates_lost,
    '606': Exception_GPIB_communication_language_lost,
    '700': Exception_Invalid_system_communication,
    '701': Exception_ASCII_only_with_RS_232,
    '800': Exception_Insufficient_vector_data,
    '801': Exception_OUTPUT_blocked_by_output_enable,
    '802': Exception_Not_permitted_with_OUTPUT_off,
    '803': Exception_Expression_list_full,
    '804': Exception_Undefined_expression_exists,
    '805': Exception_Expression_not_found,
    '806': Exception_Definition_not_allowed,
    '807': Exception_Expression_cannot_be_deleted,
    '808': Exception_Source_memory_location_revised,
    '809': Exception_OUTPUT_blocked_by_Over_Temp,
    '811': Exception_Not_an_operator_or_number,
    '812': Exception_Mismatched_parenthesis,
    '813': Exception_Not_a_number_of_data_handle,
    '814': Exception_Mismatched_brackets,
    '815': Exception_Too_many_parenthesis,
    '816': Exception_Entire_expression_not_parsed,
    '817': Exception_Unknown_token,
    '818': Exception_Error_parsing_mantissa,
    '819': Exception_Error_parsing_exponent,
    '820': Exception_Error_parsing_value,
    '821': Exception_Invalid_data_handle_index,
    '822': Exception_Too_small_for_sense_range,
    '823': Exception_Invalid_with_source_read_back_on,
    '824': Exception_Cannot_exceed_compliance_range,
    '825': Exception_Invalid_with_auto_ohms_on,
    '826': Exception_Attempt_to_exceed_power_limit,
    '827': Exception_Invalid_with_ohms_guard_on,
    '828': Exception_Invalid_on_1_amp_range,
    '829': Exception_Invalid_on_1kV_range,
    '830': Exception_Invalid_with_INF_ARM_COUNT,
    '831': Exception_Invalid_in_Pulse_Mode,
    '900': Exception_Internal_System_Error,
}